packages/botonic-nlp/src/tasks/text-classification/botonic-text-classifier.ts

```diff
@@ -0,0 +1,173 @@
+import { Tensor2D } from '@tensorflow/tfjs-node'
+import { join } from 'path'
+
+import { DatasetLoader } from '../../dataset/loader'
+import { Dataset, Sample } from '../../dataset/types'
+import { generateEmbeddingsMatrix } from '../../embeddings/embeddings-matrix'
+import { WordEmbeddingStorage } from '../../embeddings/types'
+import { LabelEncoder } from '../../encode/label-encoder'
+import { OneHotEncoder } from '../../encode/one-hot-encoder'
+import { ModelManager } from '../../model/manager'
+import { ModelEvaluation } from '../../model/types'
+import { PADDING_TOKEN, UNKNOWN_TOKEN } from '../../preprocess/constants'
+import { Preprocessor } from '../../preprocess/preprocessor'
+import { trainTestSplit } from '../../preprocess/selection'
+import {
+  Normalizer,
+  PaddingPosition,
+  Stemmer,
+  Stopwords,
+  Tokenizer,
+} from '../../preprocess/types'
+import { VocabularyGenerator } from '../../preprocess/vocabulary-generator'
+import { ModelStorage } from '../../storage/model-storage'
+import { Locale } from '../../types'
+import { unique } from '../../utils/array-utils'
+import {
+  Intent,
+  PredictionProcessor,
+} from '../text-classification/process/prediction-processor'
+import { createSimpleNN } from './models/simple-nn'
+import {
+  TEXT_CLASSIFIER_TEMPLATE,
+  TextClassifierParameters,
+} from './models/types'
+import { Processor } from './process/processor'
+import { TextClassificationConfigStorage } from './storage/config-storage'
+
+export class BotonicTextClassifier {
+  vocabulary: string[]
+  classes: string[]
+  private preprocessor: Preprocessor
+  private processor: Processor
+  private predictionProcessor: PredictionProcessor
+  private modelManager: ModelManager
+
+  constructor(readonly locale: Locale, readonly maxLength: number) {
+    this.preprocessor = new Preprocessor(locale, maxLength)
+  }
+
```

Comment:
try this:
```
export type ClassifierEngines = {
  preprocessor : Preprocessor,
  textClassification: BotonicTextClassifier, 
  config: TextClassificationConfigStorage
}

class BotonicTextClassifier {
engines: ClassifierEngines
constructor( 
     readonly locale: Locale, readonly maxLength: number,
     engines: Partial<ClassifierEngines> = {}
     )
{
this.engines = {
   preprocessor : engines.preprocessor ?? new Preprocessor(locale, maxLength),
  ....
}
}
```
In the future we should be able to inject also the Processor and others and ModelManager, but let's start with this

Reply:
or
```
export type ClassifierOptions = {
  locale: Locale
  maxLength: number
  preprocessor?: Preprocessor,
  textClassification?: BotonicTextClassifier, 
  config?: TextClassificationConfigStorage
}

class BotonicTextClassifier {
options: Required<ClassifierOptions>
constructor( 
     options: ClassifierOptions = {}
     )
{
this.options = {
  locale: options.locale
  maxLength: options.maxLength
   preprocessor : options.preprocessor ?? new Preprocessor(locale, maxLength),
  ....
}
}
```

Reply:
I think it would be better to apply these changes in a new PR when modifying the `BotonicNer` and `BotonicTextClassifier` classes. I opened a [ticket](https://linear.app/hubtype/issue/BOT-329/botonictextclassifier-and-botonicner-options) for this task.